src/main/drivers/serial_softserial.c

```diff
@@ -378,12 +385,18 @@ void processTxState(softSerial_t *softSerial)
             serialInputPortDeActivate(softSerial);
             serialOutputPortActivate(softSerial);
 
-            // Start sending on next bit timing, as port manipulation takes time,
-            // and continuing here may cause bit period to decrease causing sampling errors
-            // at the receiver under high rates.
-            // Note that there will be (little less than) 1-bit delay; take it as "turn around time".
-            // XXX We may be able to reload counter and continue. (Future work.)
-            return;
+            // Start transmitting immediately if SERIAL_BIDIR_TX_PD port option is specified
```

Comment:
This is causing the glitch ... `serialOutputPortActivate` sets pin to PP mode, last port write was 1 (stopbit), pin is high until `setTxSignal`.

It would be better to set pin to IDLE/1 before `serialOutputPortActivate` (or set it during initialization, only first transmission is affected) and transmit stopbit before first byte (return here).
That way no 0x00 byte is necessary before VTX packet (only stopbit from 0x00 is visible anyway) and it won't interfere with normal mode (bidir with pullup)

Reply:
Attempted to prevent the adding of 0x00 byte, but this broke hardware uart mode.

Reply:
No stop bit was generated using hardware uart, so the first rising edge was in the middle of the first byte.

Reply:
I see ... 
I don't see reasonable serial-level fix, everything I can think of is too complicated to be justified.

I would just transmit stopbit in softserial (to avoid glitch in pulldown mode / keep current behavior for other drivers) and add 0x00 in smartaudio (it is recommended in smartaudio documentation, so smartaudio shall handle it gracefully).

Reply:
How about setting the pin low upon deactivation of output mode when TX_PD is required?

Reply:
It's 'complicated' ... softserial is already terrible nightmare, it's bad idea to add unnecessary logic into it.
And it will generate glitch if there is pullup on other side.

You can set output to MARK before changing direction to output and start transmitting immediately, but timing issues in this code  comment may be true. 

Sending stopbit first (possibly a bit shortened, falling edge is importnt) is consistent with serial framing.

Cleanest fix would be to implement stopbit transmission in USART code, but IMO it's not worth the effort

Reply:
SmartAudio only handles the 0x00 byte 'gracefully' if there is not a full stop bit preceding the byte.  Not proceeding immediately to TX transport after changing modes by returning will generate a full stop and start bit (due to the extra cycle), which causes the unify to seemingly interpret the 0x00 byte as the first byte of the header.

However, I'm finding that the glitch is generally benign in nature and is being ignored by the Unify (although, I can't say for sure how other non-TBS hardware will interpret this).

Having this stop bit would likely be desirable for other hardware, so I'd like to propose keeping the TX_PD port option around to handle this exception.

Reply:
the RX_NOPULL *could* be used instead, but it's not directly related to the problem at hand.  It might lead to undesirable behavior if another piece of hardware requires NOPULL and doesn't require TX_PD.

Reply:
I found a way to handle the glitch.. will push soon.